pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -466,39 +470,32 @@ func ExtractMsgDetails(msg *dns.Msg) (qname string, responseIPs []net.IP, TTL ui
 // Note: This mimics what the dns package does EXCEPT for setting reuseport.
 // This is ok for now but it would simplify proxy management in the future to
 // have it set.
-func bindToAddr(address string, port uint16) (UDPConn *net.UDPConn, TCPListener *net.TCPListener, err error) {
+func bindToAddr(address string, port uint16) (*net.UDPConn, *net.TCPListener, error) {
+	var err error
+	var listener net.Listener
+	var conn net.PacketConn
 	defer func() {
-		if err == nil {
-			return
-		}
-		if TCPListener != nil {
-			TCPListener.Close()
-			TCPListener = nil
-		}
-		if UDPConn != nil {
-			UDPConn.Close()
-			UDPConn = nil
+		if err != nil {
+			if listener != nil {
+				listener.Close()
+			}
+			if conn != nil {
+				conn.Close()
+			}
 		}
 	}()
 
 	bindAddr := net.JoinHostPort(address, strconv.Itoa(int(port)))
 
-	TCPAddr, err := net.ResolveTCPAddr("tcp", bindAddr)
+	listener, err = listenConfig(0xb00).Listen(context.Background(), "tcp", bindAddr)
 	if err != nil {
 		return nil, nil, err
 	}
-	TCPListener, err = net.ListenTCP("tcp", TCPAddr)
-	if err != nil {
-		return nil, nil, err
-	}
-	UDPAddr, err := net.ResolveUDPAddr("udp", TCPListener.Addr().String())
-	if err != nil {
-		return nil, nil, err
-	}
-	UDPConn, err = net.ListenUDP("udp", UDPAddr)
+
+	conn, err = listenConfig(0xb00).ListenPacket(context.Background(), "udp", listener.Addr().String())
```

Comment:
please add documentation for the meaning of `0xb00`

Reply:
ditto